nix/pkgs/haskell/default.nix

```diff
@@ -34,27 +33,39 @@ let
   compiler-nix-name = "ghc810220201118";
 
   # The haskell project created by haskell-nix.stackProject'
-  project = import ./haskell.nix {
-    inherit lib stdenv haskell-nix buildPackages R rPackages z3;
-    inherit agdaWithStdlib checkMaterialization compiler-nix-name gitignore-nix;
-    inherit enableHaskellProfiling deferPluginErrors;
-  };
+  baseProject =
+    { deferPluginErrors }:
+    import ./haskell.nix {
+      inherit lib stdenv haskell-nix buildPackages R rPackages z3;
+      inherit agdaWithStdlib checkMaterialization compiler-nix-name gitignore-nix;
+      inherit enableHaskellProfiling;
+      inherit deferPluginErrors;
+    };
+  project = baseProject { deferPluginErrors = false; };
+  # The same as above, but this time with we defer plugin errors so that we
+  # can build "all" (the interesting) haddocks that would otherwise fail.
+  projectAllHaddock = baseProject { deferPluginErrors = true; };
 
   # All the packages defined by our project, including dependencies
   packages = project.hsPkgs;
 
   # Just the packages in the project
-  projectPackages = haskell-nix.haskellLib.selectProjectPackages packages
+  basePackages =
+    { packages }:
```

Comment:
I'd probably just go with `packages: ...`. Passing a single-argument attribute set is often overkill!

Reply:
@michaelpj do you mean make it this:

```
  # Just the packages in the project
  basePackages =
    packages:
    haskell-nix.haskellLib.selectProjectPackages packages
    # Need to list this manually to work around https://github.com/input-output-hk/haskell.nix/issues/464
    // { inherit (packages) plutus-metatheory; };
```

?

because this results in an error (and I'm not totally sure why ...):

```
error: attribute 'plutus-metatheory' missing, at /home/noon/dev/tweag/io/plutus/nix/pkgs/haskell/default.nix:57:9
```

Reply:
Well now you need to change the call sites too, i.e. instead of `basePackages { inherit packages = foo; }` just `basePackages foo`. I think this might just be confusing syntax: `x: body` is the lambda syntax, `{ x }: body` is a lambda which takes an attribute set as an argument and pattern matches on it for the fields. We often use single-attribute-set-argument lambdas to give us named arguments, but for simple cases you can just do boring lambdas with positional arguments.

Reply:
Of course :facepalm: thanks :)